skimage/segmentation/_seeds.pyx

```diff
@@ -0,0 +1,725 @@
+            l0n = (l0_hist[i] - block_hist[0]) * l0_scale
+            # Normalised histogram of label `
+            l1n = l1_hist[i] * l1_scale
+            # Normalised histogram of block
+            bn = block_hist[i] * b_scale
+            # Intersections
+            l0int += min(l0n, bn)
+            l1int += min(l1n, bn)
+
+        # Score = change in intersection
+        return l1int - l0int
+
+    return 0.0
+
+
+@cython.boundscheck(False)
+@cython.cdivision(True)
+cdef double pixel_relabel_score(int pixel_index_value,
+                                int label_0, int[:] l0_hist, int l0_area,
+                                int label_1, int[:] l1_hist, int l1_area):
+    """
+    Compute the improvement obtained by changing the label of a pixel from
+    `label_0` to `label_1`.
+
+    The score is obtained by first removing the contribution of the pixel
+    from label 0's histogram. The intersection of the histogram of the pixel
+    with the histograms of label 0 and label ` are computed and the difference
+    returned as the score.
+
+    Parameters
+    ----------
+    pixel_index_value : the index (histogram bin index) of the pixel
+    label_0 : the current label of the pixel
+    l0_hist : the histogram of the pixels labelled `label_0`
+    l0_area : the number of pixels labelled `label_0`
+    label_1 : the proposed label of the pixel
+    l1_hist : the histogram of the pixels labelled `label_1`
+    l1_area : the number of pixels labelled `label_1`
+
+    Returns
+    -------
+    A score that measures the improvement, or 0.0 if `label_0 == label_1`.
+    """
+    cdef double l0_scale = 1.0, l1_scale = 1.0, l0n = 0.0, l1n = 0.0, l0int=0.0, l1int=0.0;
+
+    if label_0 != label_1:
+        l0_scale = 1.0 / (l0_area - 1)
+        l1_scale = 1.0 / l1_area
+
+        l0n = (l0_hist[pixel_index_value] - 1) * l0_scale
+        l1n = l1_hist[pixel_index_value] * l1_scale
+        l0int += min(l0n, 1)
+        l1int += min(l1n, 1)
+
+        return l1int - l0int
+
+    return 0.0
+
+
+
+@cython.boundscheck(False)
+cdef bint _relabel_check(int a0, int a1, int b0, int b1, int c0, int c1):
+    """
+    Check if re-labelling a block or pixel will break a labelled region
+    into disjoint sub-regions.
+
+    An explanation; assume that we are relabelling a pixel to match that
+    of its neighbour above.
+    The pixel that is to be relabelled is represented by `b1`.
+    The pixel behind/below it is `b0`.
+    The pairs `a0`/`a1` and `c0`/`c1` represent pixels either side, e.g.
+    `a0` represents the pixel to the left and behind, while `a1` represents
+    the pixel to the right. `c0`/`c1` represent pixels to the right.
+
+    Parameters
+    ----------
+    a0, a1, b0, b1, c0, c1 : the labels of pixels/blocks in a neighbourhood
+    surrounding the pixel/block that is to be relabelled
+
+    Returns
+    -------
+    A boolean indicating if the proposed relabelling would cause a break.
+    """
```

Comment:
Think about how your calls will read when designing an API. You want it to be as close as possible to English. In this case, for example, you could call it `_relabel_is_safe`, or you could reverse the boolean result and call it `_relabel_will_disconnect`.